### PR TITLE
CBA extended loadout support

### PR DESCRIPTION
**Pull request by BrettMayson**

### CODE REVIEW

File: addons/main/CfgVehicles.hpp [hunk 1]
```diff
@@ -24,14 +24,14 @@ class CfgVehicles
...
+    class CAManBase: Man
+    {
+        class ACE_SelfActions
+        {
+			class ACE_Equipment {
+                #include "ACE_SelfActions.hpp"
+			};
+		};
+	};
 };
```
Comment (DerZade) [inline addons/main/CfgVehicles.hpp]: What is this formatting? :D 
```suggestion
        class ACE_SelfActions
        {
            class ACE_Equipment {
                #include "ACE_SelfActions.hpp"
            };
        };
    };
};
```

File: addons/main/functions/misc/fn_registerCBASettings.sqf [hunk 2]
```diff
@@ -38,3 +38,21 @@
     (call compile format ["[%1]", _this]) call GRAD_slingHelmet_fnc_addHelmetToWhitelist;
   }
 ] call CBA_Settings_fnc_init;
+
+["CBA_loadoutSet", {
+    params ["_unit", "_loadout", "_extradata"];
+    private _helmetClass = _extradata getOrDefault ["grad_slingHelmet", ""];
+    if (_helmetClass != "") then {
```
Comment (DerZade) [inline addons/main/functions/misc/fn_registerCBASettings.sqf]:
```suggestion
    if (_helmetClass isNotEqualTo "") then {
```

File: addons/main/functions/misc/fn_registerCBASettings.sqf [hunk 3]
```diff
@@ -38,3 +38,21 @@
...
+        [_unit, _helmetClass] call GRAD_slingHelmet_fnc_addSlungHelmet;
+    } else {
+        [_unit] call GRAD_slingHelmet_fnc_removeSlungHelmet;
+    };
+}] call CBA_fnc_addEventHandler;
+
+["CBA_loadoutGet", {
+    params ["_unit", "_loadout", "_extradata"];
+    private _helmetClass = [_unit] call GRAD_slingHelmet_fnc_getSlungHelmet;
+    if (_helmetClass != "") then {
```
Comment (DerZade) [inline addons/main/functions/misc/fn_registerCBASettings.sqf]:
```suggestion
    if (_helmetClass isNotEqualTo "") then {
```

**Review by DerZade — CHANGES_REQUESTED** [review]

**Comment by DerZade** [comment]
Thanks a lot for the contribution 😇

Hope I get to release a new version in the near future :)

File: addons/main/CfgVehicles.hpp (hunk 1 above)
Comment (BrettMayson) [reply addons/main/CfgVehicles.hpp]: /shrug

Spacing was all off and my editor was angry, so I fixed it

**Comment by BrettMayson** [comment]
[Untitled.webm](https://user-images.githubusercontent.com/3946739/190374773-4200bc48-3658-4972-a94f-9667b3dcfe99.webm)

Example of using this with the ACE arsenal to load a loadout with a saved slung helmet.

File: addons/main/functions/misc/fn_registerCBASettings.sqf [hunk 4]
```diff
@@ -38,3 +38,21 @@
...
+    private _helmetClass = _extradata getOrDefault ["grad_slingHelmet", ""];
+    if (_helmetClass isNotEqualTo "") then {
+        [_unit, _helmetClass] call GRAD_slingHelmet_fnc_addSlungHelmet;
+    } else {
+        [_unit] call GRAD_slingHelmet_fnc_removeSlungHelmet;
+    };
+}] call CBA_fnc_addEventHandler;
+
+["CBA_loadoutGet", {
+    params ["_unit", "_loadout", "_extradata"];
```
Comment (DerZade) [inline addons/main/functions/misc/fn_registerCBASettings.sqf]: To fix SQFLint error:
```suggestion
    params ["_unit", "", "_extradata"];
```

File: addons/main/functions/misc/fn_registerCBASettings.sqf [hunk 5]
```diff
@@ -38,3 +38,21 @@
     (call compile format ["[%1]", _this]) call GRAD_slingHelmet_fnc_addHelmetToWhitelist;
   }
 ] call CBA_Settings_fnc_init;
+
+["CBA_loadoutSet", {
+    params ["_unit", "_loadout", "_extradata"];
```
Comment (DerZade) [inline addons/main/functions/misc/fn_registerCBASettings.sqf]: To fix SQFLint error:
```suggestion
    params ["_unit", "", "_extradata"];
```

**Review by DerZade — CHANGES_REQUESTED** [review]
I tried to fix the linter issues myself, but sadly I can't commit directly to your branch 😅 

If the linter is happy after these changes, then I'm happy 😇

**Review by DerZade — APPROVED** [review]

**Comment by DerZade** [comment]
I will test the latest version ingame in the next few days and if that works well I will merge this PR :D

Thanks again!

**Pull request merged**